M/mousetrap/build_tarballs.jl

```diff
@@ -0,0 +1,60 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "mousetrap"
+version = v"0.3.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/Clemapfel/mousetrap.git", "ffa28d0bd569118320a6bb063286edfb35fc0429"),
+    GitSource("https://github.com/Clemapfel/mousetrap_julia_binding.git", "6b838ea238e118d694ffc1b3a1e0225441c8cfb3"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+echo -e "[binaries]\ncmake='/usr/bin/cmake'" >> cmake_toolchain_patch.ini
+cd mousetrap
+install_license LICENSE
+meson setup build --cross-file=$MESON_TARGET_TOOLCHAIN --cross-file=../cmake_toolchain_patch.ini
+meson install -C build
+cd ../mousetrap_julia_binding
+meson setup build --cross-file=$MESON_TARGET_TOOLCHAIN --cross-file=../cmake_toolchain_patch.ini -DJulia_INCLUDE_DIRS=$prefix/include/julia
+meson install -C build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = Platform[]
+include("../../L/libjulia/common.jl")
+for version in [v"1.7.0", v"1.8.2", v"1.9.0", v"1.10", v"1.11"]
+    for platform in libjulia_platforms(version)
+        if nbits(platform) != 32
+            push!(platforms, platform)
+        end
+    end
+end
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libmousetrap", :mousetrap),
+    LibraryProduct("libmousetrap_julia_binding", :mousetrap_julia_binding)
+]
+
+x11_platforms = filter(p -> Sys.islinux(p) || Sys.isfreebsd(p), platforms)
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("GLEW_jll")
+    Dependency("GLU_jll"; platforms = x11_platforms)
+    Dependency("GTK4_jll")
+    Dependency("libadwaita_jll")
+    Dependency("OpenGLMathematics_jll")
+    Dependency("libcxxwrap_julia_jll")
+    BuildDependency("libjulia_jll")
+    BuildDependency("Xorg_xorgproto_jll"; platforms = x11_platforms)
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.7", preferred_gcc_version = v"12.1.0")
```

Comment:
Please don't use gcc 12, you should instead use the oldest compiler possible: https://github.com/JuliaPackaging/Yggdrasil/blob/7093500a13b031888e5263aa3fcf7c63490bbe2b/CONTRIBUTING.md#compatibility-tips

Reply:
I added two commits to expand for string ABIs and to use GCC 9, which is the oldest version supported. Thank you